climada/engine/impact.py

```diff
@@ -344,8 +344,8 @@ def calc_risk_transfer(self, attachment, cover):
             risk_transfer.at_event = imp_layer
             risk_transfer.aai_agg = np.sum(imp_layer * new_imp.frequency)
             return new_imp, risk_transfer
-
-        return new_imp, Impact()
+        else:
+            return self, Impact()
```

Comment:
I am not convinced by this yet This fundamentally changes what object is returned. This can have unwanted consequences.

Reply:
Actually, the method should never be called if attachment or cover is not defined.

Reply:
That's the other option. Happy to implement that.